src/mono/mono/mini/mini-runtime.c

```diff
@@ -358,6 +358,7 @@ mono_print_method_from_ip (void *ip)
  */
 gboolean mono_method_same_domain (MonoJitInfo *caller, MonoJitInfo *callee)
 {
+#ifndef ENABLE_NETCORE
```

Comment:
(Same problem in #46346 ) this `#ifndef` should be lower, before the line `cmethod = jinfo_get_method (caller);` - we still want to return FALSE for trampolines and other edge cases

Reply:
Fix like this: https://github.com/dotnet/runtime/pull/46346/commits/fed6bcc2afa49822e5f1360de39d51bace82fd6c

Reply:
Yeah, will wait for you to land that first and rebase this PR